File: binary-tree/binary-tree-postorder-traversal/binary-tree-postorder-traversal.js
function TreeNode(val, left, right) {
    this.val = (val === undefined ? 0 : val)
    this.left = (left === undefined ? null : left)
    this.right = (right === undefined ? null : right)
}
/**
 * @param {TreeNode} root
 * @return {number[]}
 */
var postorderTraversal = function (root) {
    let arr = [];

    (function (root) {
        if (root) {
            arguments.callee(root.left);
            arguments.callee(root.right);
            arr.push(root.val);
        }
    })(root)

    return arr;
};

File: binary-tree/binary-tree-postorder-traversal/binary_tree_postorder_traversal.py
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def postorderTraversal(self, root: TreeNode) -> List[int]:
        arr = []

        def postorder(root:TreeNode):
            if root:
                postorder(root.left)
                postorder(root.right)
                arr.append(root.val)
        
        postorder(root)
        return arr
        